src/test/kotlin/ar/edu/unahur/obj2/servidorWeb/ServidorWebTest.kt

```diff
@@ -7,29 +7,83 @@ import java.time.LocalDateTime
+    servidor.agregarAnalizador(analizador1)
+    servidor.agregarAnalizador(analizador2)
+    servidor.agregarAnalizador(analizador3)
 
     it("devuelve 501 si recibe un pedido que no es HTTP") {
-      val respuesta = servidor.realizarPedido("207.46.13.5", "https://pepito.com.ar/hola.txt", LocalDateTime.now())
+      val respuesta = servidor.realizarPedido(Pedido("207.46.13.5", "https://pepito.com.ar/hola.txt", LocalDateTime.now()))
       respuesta.codigo.shouldBe(CodigoHttp.NOT_IMPLEMENTED)
       respuesta.body.shouldBe("")
     }
 
     it("devuelve 200 si algún módulo puede trabajar con el pedido") {
-      val respuesta = servidor.realizarPedido("207.46.13.5", "http://pepito.com.ar/hola.txt", LocalDateTime.now())
+      val respuesta = servidor.realizarPedido(Pedido("207.46.13.5", "http://pepito.com.ar/hola.txt", LocalDateTime.now()))
       respuesta.codigo.shouldBe(CodigoHttp.OK)
       respuesta.body.shouldBe("todo bien")
     }
 
     it("devuelve 404 si ningún módulo puede trabajar con el pedido") {
-      val respuesta = servidor.realizarPedido("207.46.13.5", "http://pepito.com.ar/playa.png", LocalDateTime.now())
+      val respuesta = servidor.realizarPedido(Pedido("207.46.13.5", "http://pepito.com.ar/playa.png", LocalDateTime.now()))
       respuesta.codigo.shouldBe(CodigoHttp.NOT_FOUND)
       respuesta.body.shouldBe("")
     }
+
+    it("devuelve la cantidad de respuestas demoradas para el modulo1") {
+      servidor.realizarPedido(Pedido("207.46.13.5", "http://pepito.com.ar/hola.txt", LocalDateTime.now()))
+      analizador1.respuestasDemoradasPara(modulo1).shouldBe(1)
+    }
+
+    it("devuelve la cantidad de pedidos que realizó una IP sospechosa") {
+      servidor.realizarPedido(Pedido("207.46.13.5", "http://pepito.com.ar/hola.txt", LocalDateTime.now()))
+      servidor.realizarPedido(Pedido("207.46.13.5", "http://pepito.com.ar/hola.jpg", LocalDateTime.now()))
+      analizador2.cuantosPedidosRealizo("207.46.13.5").shouldBe(2)
+    }
+
+    it("devuelve cuál fue el módulo más consultado por todas las IPs sospechosas") {
+      servidor.realizarPedido(Pedido("207.46.13.5", "http://pepito.com.ar/hola.txt", LocalDateTime.now()))
+      servidor.realizarPedido(Pedido("207.46.13.4", "http://pepito.com.ar/hola.jpg", LocalDateTime.now()))
+      servidor.realizarPedido(Pedido("207.46.13.5", "http://pepito.com.ar/hola.txt", LocalDateTime.now()))
+      analizador2.moduloMasConsultadoPorIPs().shouldBe(modulo1)
+    }
+
+    it("devuelve el conjunto de IPs sospechosas que requirieron una ruta") {
+      servidor.realizarPedido(Pedido("207.46.13.5", "http://pepito.com.ar/documentos/doc1.txt", LocalDateTime.now()))
+      servidor.realizarPedido(Pedido("207.46.13.4", "http://pepito.com.ar/documentos/doc1.txt", LocalDateTime.now()))
+      servidor.realizarPedido(Pedido("207.46.13.3", "http://pepito.com.ar/documentos/doc1.txt", LocalDateTime.now()))
+      analizador2.ipsQueRequirieronLaRuta("documentos/doc1.txt").count().shouldBe(3)
```

Comment:
Sería mejor chequear que efectivamente te devuelve las IPs que querés.

Reply:
No se me ocurrió como realizar eso, que me devolviera los elementos que tiene adentro y me los mostrara en pantalla. Por eso usé el count

Reply:
Lo hacés así:

```suggestion
      analizador2.ipsQueRequirieronLaRuta("documentos/doc1.txt").shouldContainExactly("207.46.13.5", "207.46.13.4", "207.46.13.3")
```

En la documentación tenés todos los _matchers_ que se banca kotest: https://github.com/kotest/kotest/blob/master/doc/matchers.md